OpenRA.Game/Primitives/Polygon.cs

```diff
@@ -0,0 +1,114 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+
+namespace OpenRA.Primitives
+{
+	public struct Polygon
+	{
+		public static readonly Polygon Empty = new Polygon(Rectangle.Empty);
+
+		public readonly Rectangle BoundingRect;
+		public readonly int2[] Vertices;
```

Comment:
I believe it is best to leave this as is for performance. I just want to note that code will have to be audited for changes to vertices after construction, since the bounding rectangle and rectangle and rectangle flag would fall out of sync.

Reply:
Maybe i'm missing something, but both `Rectangle` and `int2` are immutable so this shouldn't be possible?